pkg/dnssort/tree_test.go

```diff
@@ -0,0 +1,60 @@
+package dnssort
+
+import "testing"
+
+func Test_domaintree(t *testing.T) {
+
+	t.Run("Single FQDN",
+		executeTreeTest(
+			[]string{
+				"other.example.com",
+			},
+			[]string{"other.example.com"},
+			[]string{"com", "x.example.com", "x.www.example.com", "example.com"},
+		),
+	)
+
+	t.Run("Wildcard",
+		executeTreeTest(
+			[]string{
+				"*.example.com",
+			},
+			[]string{"example.com", "other.example.com"},
```

Comment:
Missing, I will add it to the test case. the wildcard is only resolved when adding things to the tree, not when looking them up. This is on purpose because the record dependencies never contain wildcards, while record names can.